java/c/src/test/python/integration_tests.py

```diff
@@ -190,6 +190,17 @@ def round_trip_reader(self, schema, batches):
     def test_string_array(self):
         self.round_trip_array(lambda: pa.array([None, "a", "bb", "ccc"]))
 
+    def test_string_slice_array(self):
+        data = pa.array(["foo", "bar", "baz1", "baz223", "baz23445", "baz2121", "12312baz"])
```

Comment:
Can we include some null and empty strings? (Both before and after the slice point)

Reply:
I'd also like to see a case where the sliced array is empty.

Reply:
Sure, let me add those cases.